### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by aichholzer** [redcenter/api-design-tips#6]
Every good API must start at the place; documentation.
API designers, architects and developers tend to forget about this crucial part and information can later only be found, scattered about, in emails, text files, gists and what not.

Documentation should be treated as the most important part in any API specification and implementation. If you don't have a well created document then you will, eventually, end up answering questions on how this or that works.

Here's a tool I suggest to use (and mention in in your tips): http://ab.je/7r
and it's specification: http://ab.je/7s

**Comment by redcenter** [comment]
Thanks for all your great ideas. I will look into them.

Please remember: like I said in the document, these are tips based on my own experiences. I will add suggestions when I have used them myself first, if that's ok with you...

If you have an article/repo yourself on this topic, I would love to link to it. Again thanks.

**Comment by aichholzer** [comment]
You are welcome. I did not mean, in any way, to come across like a smart-ass, just trying to make your document better. I have been thinking about writing something like this for a while but can't quite find the time, will do though.

**Comment by redcenter** [comment]
Hi, thanks for you input. I have added some of your suggestions and mentioned you!

About your initial issues: I know you don't mean to be a smart-ass and you are definitely not. But always remember to use the "communication sandwich" approach in dealing in this sort of situations. 

- First start with a compliment
- Make your point 
- End with a compliment

That will take the edge off your comment and people are more inclined to take you seriously and have a good feeling about your issue. Especially on internet and in peer reviews between total strangers, it's essential for inter-personal relationships... :-)
 
Of course you only do this on the first issue, the others can be to-the-point... :-)

Again thanks for your suggestions and all the time you put in my document! Really appreciate it.

Greetings, Jeroen

PS see the sandwich approach in the responses I gave you?  ;-)
